### PR TITLE
feat: session auth refactor

### PR DESCRIPTION
**Pull request by porcellus**
## Summary of change

- Cleaned up some types of the session context
- Removed `redirectToLogin` from `SessionAuth` (it was in the changelog and unused but not removed, this change was lost somewhere)
- Added `useClaimValue` + added it into a test

## Related issues

-   

## Test Plan

Updated a test that includes it.

## Documentation changes

Done separately

## Checklist for important updates

-   [x] Changelog has been updated
-   [x] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `lib/ts/version.ts`
-   [x] Changes to the version if needed
    -   In `package.json`
    -   In `package-lock.json`
    -   In `lib/ts/version.ts`
-   [x] Had run `npm run build-pretty`
-   [x] Had installed and ran the pre-commit hook
-   [x] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.
-   [x] If added a new recipe interface, then make sure that the implementation of it uses NON arrow functions only (like `someFunc: function () {..}`).
-   [x] If I added a new recipe, I also added the recipe entry point into the `size-limit` section of `package.json` with the size limit set to the current size rounded up.

### CODE REVIEW

File: lib/ts/recipe/session/sessionContext.ts [hunk 1]
```diff
@@ -1,14 +1,9 @@
 import React from "react";
 import { SessionContextType } from "./types";
 
-const SessionContext = React.createContext<SessionContextType & { isDefault: boolean }>({
+const SessionContext = React.createContext<SessionContextType & { isDefault?: boolean }>({
```
Comment (rishabhpoddar) [inline lib/ts/recipe/session/sessionContext.ts:4]: why is isDefault optional? I thought that it will be used by SessionAuth to know that if it's already enclosed by a parent SessionAuth or not, and also that the user exposed context type will not have this. So for internal use, shouldn't we keep isDefault compulsory?

File: lib/ts/recipe/session/sessionAuth.tsx [hunk 2]
```diff
@@ -184,9 +179,7 @@ const SessionAuth: React.FC<PropsWithChildren<SessionAuthProps>> = ({ children,
         return null;
     }
 
-    return (
-        <SessionContext.Provider value={{ ...context, isDefault: false } as any}>{children}</SessionContext.Provider>
-    );
+    return <SessionContext.Provider value={context}>{children}</SessionContext.Provider>;
```
Comment (rishabhpoddar) [inline lib/ts/recipe/session/sessionAuth.tsx:182]: why are we not giving isDefault as false anymore?

File: lib/ts/recipe/session/useSessionContext.ts [hunk 3]
```diff
@@ -6,7 +6,7 @@ import { SessionContextType } from "./types";
 const useSessionContext = (): SessionContextType => {
     const ctx = React.useContext(SessionContext);
 
-    if (ctx.isDefault) {
+    if (ctx.isDefault === true) {
         throw new Error("Cannot use useSessionContext outside auth wrapper components.");
     }
```
Comment (rishabhpoddar) [inline lib/ts/recipe/session/useSessionContext.ts:11]: why this change? Cause not if isDefault is undefined, it will not throw the error - and from the types, it can be undefined. When will it be undefined?

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: lib/ts/recipe/session/sessionAuth.tsx (hunk 2 above)
Comment (porcellus) [reply lib/ts/recipe/session/sessionAuth.tsx:182]: We are checking for strict equality anyway and this way we don't need the `as any` to make the typing work.

File: lib/ts/recipe/session/sessionContext.ts (hunk 1 above)
Comment (porcellus) [reply lib/ts/recipe/session/sessionContext.ts:4]: - it makes typing it properly in `SessionAuth` easier (it could be done I think, but more work)
- we are checking strict equality anyway.

File: lib/ts/recipe/session/useSessionContext.ts (hunk 3 above)
Comment (porcellus) [reply lib/ts/recipe/session/useSessionContext.ts:11]: It's undefined if it's provided by `SessionAuth` (after the change). We only need that prop in the default/initialization value of the context. We could set it to false everywhere else but just omitting it is fine since we never actually want to set it to true.

**Review by rishabhpoddar — APPROVED** [review]
